### ISSUE TITLE
KeymanagementService - ListKeys - No baseUri/endpoint parameter

### ISSUE THREAD

**Issue by jbcintra** [oracle/oci-dotnet-sdk#61]
I'm trying to load keys out via ListKeys.  I'm passing in the Compartment Id.
The output is an error "Value cannot be null. Parameter 'baseUri'.  This does not exist in the C# SDK definitions or methods.

From testing in the CLI, it specifies passing in the --endpoint variable (CryptoEndpoint property in Vault) from the parent Vault, so I would expect to ListVaults, enumerate the vaults and call ListKeys for each one passing in Compartment & Endpoint parameter.

**Comment by github-anurag** [comment]
Hi @jbcintra 
Thank you for reaching out.
An example for using ListKeys can be found at: https://docs.oracle.com/en-us/iaas/api/#/en/key/release/KeySummary/ListKeys 
If you still see the issue can you provide the following:-
1. The .NET SDK version you are using?
2. A code snippet.

**Comment by jbcintra** [comment]
Hi @github-anurag 

I'm using the latest SDK, updated this week.  The issue is running code to query keys ... fails and asks for another parameter that is not defined in the Model, nor in the examples, so no means to pass it in.

Simple code passing only CompartmentId:
public IEnumerable<KeySummary> ListKeys(string compartmentId, KmsManagementClient client)
		{
			var request = new ListKeysRequest() {CompartmentId = compartmentId };
			var response = client.ListKeys(request);
			return response.Result.Items;
		}

Then using the CLI against the API, it raises an error unless the parameter --endpoint is included e.g.
oci kms management key list --compartment-id <ocid>

with or without "--all" triggers the missing variable "--endpoint"

From the Vaults output, passing in the ManagementEndpoint as the parameter --endpoint runs properly. (CryptoEndpoint passed in returns 404)

**Comment by github-anurag** [comment]
Hi @jbcintra 

You can set the endpoint you want to use while creating the client by using the following:-
```
var client = new KmsManagementClient(provider, new ClientConfiguration(), "your-endpoint");
```
Let me know if this works for you.

**Comment by jbcintra** [comment]
Hi @github-anurag 
Thanks for the pointer. I've had to rejig the code to create the KmsManagementClient just in time with provider, clientConfig, and endpoint.

It makes the code more cumbersome having to create new clients constantly, unlike those I've created for the rest of the OCI stack (~50), this is the only one with that requirement.  It's an issue as I'm auditing multiple tenancies and multiple vaults, so have to enumerate the tenancies and vaults, then create completely new KmsManagementClients on every pass.  But ... it's working.

Is this element going to be moved into being a parameter to align with all the other OciClients?

Thanks,

Justin.
